Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/IncreasingRadius/ResolveSceneStateByIncreasingRadiusSystem.cs

```diff
@@ -96,117 +161,268 @@ private void ProcessScenesUnloadingInRealm()
+                int orderedDataNativeLength = orderedDataNative.Length;
+                OrderedDataNative* dataPtr = orderedDataNative.GetUnsafePtr();
+                int xCoordinateTeleporting = teleportParcel.Parcel.x;
+                int yCoordinateTeleporting = teleportParcel.Parcel.y;
 
-                // As sorting is throttled Entity might gone out of scope
-                if (!World.IsAlive(data.Entity))
-                    continue;
+                for (var i = 0; i < orderedDataNativeLength && promisesCreated < realmPartitionSettings.ScenesRequestBatchSize; i++)
+                {
+                    OrderedDataManaged data = orderedDataManaged[dataPtr[i].ReferenceListIndex];
 
-                // We can't save component to data as sorting is throttled and components could change
-                var components
-                    = World.Get<SceneDefinitionComponent, PartitionComponent, VisualSceneState>(data.Entity);
+                    //Ignore unpartitioned and out of range
+                    //Optimization: remove out of range from list when adding DeleteEntityIntention
+                    if (data.RawSqrDistance < 0 || data.OutOfRange) continue;
+
+                    if (teleportParcel.IsTeleporting)
+                    {
+                        if (data.Vector2IntGridLookup.Contains(xCoordinateTeleporting, yCoordinateTeleporting))
+                        {
+                            UpdateLoadingState(ipfsRealm, data.Entity, data.SceneDefinitionComponent, data.PartitionComponent, data.SceneLoadingState);
+                            break;
+                        }
+                        continue;
+                    }
+
+                    UpdateLoadingState(ipfsRealm, data.Entity, data.SceneDefinitionComponent, data.PartitionComponent, data.SceneLoadingState);
+                }
+            }
+
+
+        }
+
+        private void TryUnload(in Entity entity, ref SceneLoadingState sceneState)
+        {
+            if (sceneState.PromiseCreated)
+                Unload(entity, ref sceneState);
+        }
 
-                switch (components.t2.Value.CurrentVisualSceneState)
+        private void Unload(in Entity entity, ref SceneLoadingState sceneState)
+        {
+            sceneState.VisualSceneState = VisualSceneState.UNINITIALIZED;
+            sceneState.PromiseCreated = false;
+            sceneState.FullQuality = false;
+            World.Add(entity, DeleteEntityIntention.DeferredDeletion);
+        }
+
+        private void UpdateLoadingState(IIpfsRealm ipfsRealm, in Entity entity, in SceneDefinitionComponent sceneDefinitionComponent, in PartitionComponent partitionComponent,
+            SceneLoadingState sceneState)
+        {
+            VisualSceneState candidateBy
+                = visualSceneStateResolver.ResolveVisualSceneState(partitionComponent, sceneDefinitionComponent, sceneState.VisualSceneState, ipfsRealm.SceneUrns.Count > 0);
```

Comment:
You can move this piece to the job

Reply:
As discussed, no need to parallelize now. Including it in the `SortJob` would be tricky, and the benefit agaisnt the code complexity may not be worth it